centreon/src/Core/AdditionalConnectorConfiguration/Infrastructure/Repository/DbWriteAccRepository.php

```diff
@@ -0,0 +1,122 @@
+
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\AdditionalConnectorConfiguration\Application\Repository\WriteAccRepositoryInterface;
+use Core\AdditionalConnectorConfiguration\Domain\Model\Acc;
+use Core\AdditionalConnectorConfiguration\Domain\Model\NewAcc;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\Common\Infrastructure\Repository\RepositoryTrait;
+use Security\Interfaces\EncryptionInterface;
+
+class DbWriteAccRepository extends AbstractRepositoryRDB implements WriteAccRepositoryInterface
+{
+    use RepositoryTrait;
+
+    /**
+     * @param EncryptionInterface $encryption
+     * @param DatabaseConnection $db
+     */
+    public function __construct(
+        private readonly EncryptionInterface $encryption,
+        DatabaseConnection $db
+    )
+    {
+        $encryption->setSecondKey(Acc::ENCRYPTION_KEY);
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function add(NewAcc $acc): int
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                INSERT INTO `:db`.`additional_connector_configuration`
+                    (type, name, description, parameters, created_by, created_at, updated_by, updated_at)
+                VALUES (:type, :name, :description, :parameters, :createdBy, :createdAt, :createdBy, :createdAt)
+                SQL
+        ));
+
+        $statement->bindValue(':type', $acc->getType()->value, \PDO::PARAM_STR);
+        $statement->bindValue(':name', $acc->getName(), \PDO::PARAM_STR);
+        $statement->bindValue(':description', $acc->getDescription(), \PDO::PARAM_STR);
+        $statement->bindValue(':parameters', json_encode($acc->getParameters()->encryptCredentials($this->encryption)));
+        $statement->bindValue(':createdBy', $acc->getCreatedBy(), \PDO::PARAM_INT);
+        $statement->bindValue(':createdAt', $acc->getCreatedAt()->getTimestamp(), \PDO::PARAM_INT);
+        $statement->execute();
+
+        return (int) $this->db->lastInsertId();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function update(Acc $acc): void
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                UPDATE `:db`.`additional_connector_configuration`
+                SET
+                    `name` = :name,
+                    `description` = :description,
+                    `parameters` = :parameters,
+                    `updated_by` = :updatedBy,
+                    `updated_at` = :updatedAt
+                WHERE
+                    `id` = :id
+                SQL
+        ));
+
+        $statement->bindValue(':id', $acc->getId(), \PDO::PARAM_INT);
+        $statement->bindValue(':name', $acc->getName(), \PDO::PARAM_STR);
+        $statement->bindValue(':description', $acc->getDescription(), \PDO::PARAM_STR);
+        $statement->bindValue(':parameters', json_encode($acc->getParameters()));
```

Comment:
credentials are not encrypted on update? (or maybe it is handled on the PUT PR ?)